netx.go

```diff
@@ -38,14 +86,26 @@ func DialUDP(network string, laddr, raddr *net.UDPAddr) (*net.UDPConn, error) {
 func DialTimeout(network string, addr string, timeout time.Duration) (net.Conn, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	conn, err := DialContext(ctx, network, addr)
+	if isNetworkUnreachable(err) {
+		addr = convertAddressDNS64(addr)
+		conn, err = DialContext(ctx, network, addr)
+	}
+
 	cancel()
 	return conn, err
 }
 
 // DialContext dials the given addr on the given net type using the configured
 // dial function, with the given context.
 func DialContext(ctx context.Context, network string, addr string) (net.Conn, error) {
-	return dial.Load().(func(context.Context, string, string) (net.Conn, error))(ctx, network, addr)
+	dialer := dial.Load().(func(context.Context, string, string) (net.Conn, error))
+
+	conn, err := dialer(ctx, network, addr)
+	if isNetworkUnreachable(err) {
+		addr = convertAddressDNS64(addr)
```

Comment:
what if it's already an ipv6 address?

Reply:
it looks like now it will return the same address from the conversion if it's already ipv6, so in that case it's still a bit odd because it will re-dial the same address ?